web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java

```diff
@@ -111,31 +111,26 @@ public Response doGet(
         ShortestPathTree shortestPathTree = new ShortestPathTree(queryGraph, queryGraph.wrapWeighting(weighting), reverseFlow, traversalMode);
 
         double limit;
-        if (weightLimit.get() > 0) {
-            limit = weightLimit.get();
+        ToDoubleFunction<ShortestPathTree.IsoLabel> fz;
+        if (weightLimit.orElseThrow(() -> new IllegalArgumentException("query param weight_limit is not a number.")) > 0) {
```

Comment:
Under which condition will this exception be thrown? And what happened previously under the same condition? Can we add a test for this (just for one of the parameters maybe)'

Reply:
> Under which condition will this exception be thrown? And what happened previously under the same condition?

The handling for Optinal(Double/Int/Long) within dropwizard is a bit different.

For example:

* `weight_limit=1000` -> weightLimit.isPresent() == true && weightLimit.getAsLong() == 1000L
* no query parameter -> weightLimit.isPresent() == true && weightLimit.getAsLong() == -1L (populated via `@DefaultValue`)
* `weight_limit=garbage` -> weightLimit.isPresent() == false && weightLimit.getAsLong() throws NoSuchElementException

> Can we add a test for this (just for one of the parameters maybe)?

We already have such tests:

https://github.com/graphhopper/graphhopper/blob/cff273999af74eff25cdd888087b068de21020b1/web/src/test/java/com/graphhopper/application/resources/IsochroneResourceTest.java#L280

The exception messages are the same as the previous ones to avoid client breakage.

Reply:
Ok nice, so in case `weight_limit=garbage` we even get the same error message as before. And when it is missing we get the default value as before.

:+1: